index.test-d.ts

```diff
@@ -13,19 +16,35 @@ import {
 	supportedMimeTypes,
 	type FileTypeResult,
 	type ReadableStreamWithFileType,
+	FileTypeParser,
 } from './index.js';
 
 expectType<Promise<FileTypeResult | undefined>>(fileTypeFromBuffer(new Uint8Array([0xFF, 0xD8, 0xFF])));
 expectType<Promise<FileTypeResult | undefined>>(fileTypeFromBuffer(new ArrayBuffer(42)));
 
+// FileTypeFromBuffer
 (async () => {
 	const result = await fileTypeFromBuffer(new Uint8Array([0xFF, 0xD8, 0xFF]));
 	if (result !== undefined) {
 		expectType<string>(result.ext);
 		expectType<string>(result.mime);
 	}
+
+	expectType<FileTypeResult | undefined>(await fileTypeFromBuffer(new Uint8Array([0xFF, 0xD8, 0xFF]), {customDetectors: []}));
 })();
 
+// FileTypeFromBlob
+(async () => {
+	const result = await fileTypeFromBlob(new Blob());
+	if (result !== undefined) {
+		expectType<string>(result.ext);
+		expectType<string>(result.mime);
+	}
+
+	expectType<FileTypeResult | undefined>(await fileTypeFromBlob(new Blob(), {customDetectors: []}));
+})();
```

Comment:
We don't need to duplicate tests for each method. Generally, type tests are only useful if they do something more than just check what we can already see from the types. Same with the rest of the changes in this file.